.github/workflows/integration.yml

```diff
@@ -0,0 +1,385 @@
+          ${KUBECTL} apply -f "https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSION}/namespace.yaml"
+          ${KUBECTL} apply -f "https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSION}/network-addons-config.crd.yaml"
+          ${KUBECTL} apply -f "https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSION}/operator.yaml"
+
+          cat <<EOF | ${KUBECTL} apply -f -
+          apiVersion: networkaddonsoperator.network.kubevirt.io/v1
+          kind: NetworkAddonsConfig
+          metadata:
+            name: cluster
+          spec:
+            imagePullPolicy: IfNotPresent
+            linuxBridge: {}
+            multus: {}
+          EOF
+
+          echo "Waiting for CNAO to be ready"
+          ${KUBECTL} wait --for condition=Available networkaddonsconfig cluster --timeout=5m
+
+          echo "Successfully deployed CNAO:"
+          ${KUBECTL} get networkaddonsconfig cluster -o yaml
+        shell: bash
+        env:
+          CLUSTER_NAME: kv-testing
+          KIND: kind
+          KUBECTL: kubectl
+          CNAO_VERSION: v0.87.0
+
+      - name: create NAD
+        if: inputs.ansible_test_targets != ''
+        run: |
+          echo "Creating NetworkAttachmentDefinition (with bridge CNI)"
+          cat <<EOF | ${KUBECTL} apply -f -
+          apiVersion: k8s.cni.cncf.io/v1
+          kind: NetworkAttachmentDefinition
+          metadata:
+            name: ${SECONDARY_NETWORK_NAME}
+            namespace: ${NAMESPACE}
+          spec:
+            config: |
+              {
+                "cniVersion": "0.3.1",
+                "name": "${SECONDARY_NETWORK_NAME}",
+                "type": "bridge",
+                "bridge": "${SECONDARY_NETWORK_NAME}",
+                "ipam": {
+                  "type": "host-local",
+                  "ranges": [
+                    [
+                      {
+                        "subnet": "${SECONDARY_NETWORK_SUBNET}",
+                        "rangeStart": "${SECONDARY_NETWORK_RANGE_START}",
+                        "rangeEnd": "${SECONDARY_NETWORK_RANGE_END}",
+                        "gateway": "${SECONDARY_NETWORK_GATEWAY}"
+                      }
+                    ]
+                  ]
+                }
+              }
+          EOF
+
+          echo "Successfully created NetworkAttachmentDefinition:"
+          ${KUBECTL} get networkattachmentdefinition.k8s.cni.cncf.io "${SECONDARY_NETWORK_NAME}" --namespace "${NAMESPACE}" -o yaml
+        shell: bash
+        env:
+          CLUSTER_NAME: kv-testing
+          KIND: kind
+          KUBECTL: kubectl
+          SECONDARY_NETWORK_NAME: kindexgw
+          SECONDARY_NETWORK_SUBNET: 172.19.0.0/16
+          SECONDARY_NETWORK_RANGE_START: 172.19.1.1
+          SECONDARY_NETWORK_RANGE_END: 172.19.255.254
+          SECONDARY_NETWORK_GATEWAY: 172.19.0.1
+          NAMESPACE: default
```

Comment:
This is a neat idea, but I'm concerned about the duplication of the contents of `hack/e2e-setup.sh`. IMO I'd like to keep the setup in there. The actions should just use this script. This allows to use the same environment on local test runs which IMO is important.

Reply:
ok reverted to bash script